### ISSUE TITLE
Enhancement request: Ability to unshare multiple histories at once

### ISSUE THREAD

**Issue by lparsons** [galaxyproject/galaxy#3707]
I have a user who would like to clean up old data, but find it rather tedious to "unshare" many histories (which must be done before deleting).

**Comment by nsoranzo** [comment]
I think that sharing/unsharing multiple histories is tricky because there are 3 different ways to share things (by URL, completely public, with specific users).

**Comment by lparsons** [comment]
Good point. This specific case is with specific users. I don't use it since our Galaxy is behind authentication, but does one need to unshare via URL or public methods in order to delete a history?

**Comment by jennaj** [comment]
Hi @lparsons

Batch "unshare" (all share methods) seems like it might be a fit on the "Saved Histories" form. Similar to how the batch "delete/purge" buttons function. However, if a simple "unshare all" function was created (for selected histories only), it wouldn't address specific the method-sharing changes @nsoranzo brought up. Perhaps an "all" batch unshare function is better than none? Or a place to start?

And yes - histories must be unshared completely in order to delete/permanently delete them. This is true even with an admin account when done through the UI. There may be a way to do this directly with a script/API call, but I didn't find any documentation about that and have never done it myself.

@galaxyproject/guac Does anyone else know if an admin function/method to delete shared histories or to unshare in batch exists outside of the UI? Or if either has been previously considered?
